web/libs/editor/src/lib/AudioUltra/react/index.ts

```diff
@@ -29,6 +30,30 @@ export const useWaveform = (
   const [layers, setLayers] = useState<Layer[]>([]);
   const [layerVisibility, setLayerVisibility] = useState(new Map());
 
+  const onFrameChangedRef = useRef(options?.onFrameChanged);
+  onFrameChangedRef.current = options?.onFrameChanged;
+
+  const updateAfterRegionDraw = useMemo(() => {
+    let prevFrameState: WaveformFrameState | null = null;
+    let requestId = -1;
+    const updateAfterRegionDraw = (frameState: WaveformFrameState) => {
+      cancelAnimationFrame(requestId);
+      requestId = requestAnimationFrame(() => {
+        if (
+          !prevFrameState ||
+          frameState.width !== prevFrameState.width ||
+          frameState.height !== prevFrameState.height ||
+          frameState.zoom !== prevFrameState.zoom ||
+          frameState.scroll !== prevFrameState.scroll
+        ) {
+          onFrameChangedRef.current?.(frameState);
+          prevFrameState = frameState;
+        }
+      });
+    };
+    return updateAfterRegionDraw;
```

Comment:
```suggestion
    return (frameState: WaveformFrameState) => {
      cancelAnimationFrame(requestId);
      requestId = requestAnimationFrame(() => {
        if (
          !prevFrameState ||
          frameState.width !== prevFrameState.width ||
          frameState.height !== prevFrameState.height ||
          frameState.zoom !== prevFrameState.zoom ||
          frameState.scroll !== prevFrameState.scroll
        ) {
          onFrameChangedRef.current?.(frameState);
          prevFrameState = frameState;
        }
      });
    };
```
Just so it doesn't shadow the main defined const name, which can look on first glance like it is being redefined.